File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/ChampionInfoDao.java
package com.legue.axel.lolsummonertool.database.dao.champion;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.legue.axel.lolsummonertool.database.model.champion.ChampionInfo;

import java.util.List;

@Dao
public interface ChampionInfoDao {

    @Query("SELECT * FROM champion_infos ORDER BY id")
    LiveData<List<ChampionInfo>> getChampionInfos();

    @Query("SELECT * FROM champion_infos WHERE id = :championInfoId")
    LiveData<ChampionInfo> getChampionInfoById(int championInfoId);

    @Query("SELECT * FROM champion_infos WHERE championId = :championKey")
    LiveData<ChampionInfo> getChampionInfoByChampionKeyId(int championKey);

    @Insert
    void insertChampionInfo(ChampionInfo championInfo);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllChampionInfo(List<ChampionInfo> championInfoList);

    @Query("DELETE FROM champion_infos")
    void deleteAll();

    @Delete
    void deleteChampionInfo(ChampionInfo championInfo);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateChampionInfo(ChampionInfo championInfo);
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/dao/champion/ChampionInfoDao.kt
package com.legue.axel.lolsummonertool.database.dao.champion

import androidx.lifecycle.LiveData
import androidx.room.*
import com.legue.axel.lolsummonertool.database.model.champion.ChampionInfo

@Dao
interface ChampionInfoDao {
    @Query("SELECT * FROM champion_infos ORDER BY id")
    fun getChampionInfos(): LiveData<List<ChampionInfo>>

    @Query("SELECT * FROM champion_infos WHERE id = :championInfoId")
    fun getChampionInfoById(championInfoId: Int): LiveData<ChampionInfo>

    @Query("SELECT * FROM champion_infos WHERE championId = :championKey")
    fun getChampionInfoByChampionKeyId(championKey: Int): LiveData<ChampionInfo>

    @Insert
    fun insertChampionInfo(championInfo: ChampionInfo)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllChampionInfo(championInfoList: List<ChampionInfo>)

    @Query("DELETE FROM champion_infos")
    fun deleteAll()

    @Delete
    fun deleteChampionInfo(championInfo: ChampionInfo)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateChampionInfo(championInfo: ChampionInfo)
}